### PR TITLE
Add tests for anonymizer unit tests

### PR DESCRIPTION
**Pull request by tongwu-msft**
## Description
Add tests for anonymizer unit tests.

## Related issues
Addresses [issue #].

## Testing
Add unit tests.

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Anonymizer.Core.Utility;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export;
+using Microsoft.Health.Fhir.Core.Models;
+using NSubstitute;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Shared.Core.UnitTests.Features.Operations.Export
```
Comment (namadabu) [inline src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs:22]: >Shared [](http://example.com/codeflow?start=32&length=6)

namespace should be `...Fhir.Core...` and not `...Fhir.Shared.Core..`

File: src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+              ]
+            }";
+
+            Patient patient = new Patient();
+            patient.Name.Add(HumanName.ForFamily("input"));
+            IAnonymizer anonymizer = await CreateAnonymizerFromConfigContent(configurationContent);
+
+            ResourceElement resourceElement = anonymizer.Anonymize(new ResourceElement(patient.ToTypedElement()));
+            Patient anonymizedResource = resourceElement.Instance.ToPoco<Patient>();
+            string resourceId = EncryptUtility.DecryptTextFromBase64WithAes(anonymizedResource.Id, Encoding.UTF8.GetBytes("704ab12c8e3e46d4bea600ef62a6bec7"));
```
Comment (namadabu) [inline src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs:148]: >string resourceId = EncryptUtility.DecryptTextFromBase64WithAes(anonymizedResource.Id, Encoding.UTF8.GetBytes("704ab12c8e3e46d4bea600ef62a6bec7")); [](http://example.com/codeflow?start=12&length=147)

nit: is this line necessary for this test?

Comment (tongwu-msft) [reply src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs]: yes, removed.

File: src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                  ""method"": ""generalize"",
+                  ""cases"": {
+                    ""$this <= @2020-01-01 and $this >= @1990-01-01"": ""@2000-01-01""
+                  }
+                }
+              ]
+            }";
+
+            Patient patient = new Patient();
+            patient.BirthDate = "2001-01-01";
```
Comment (namadabu) [inline src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs:67]: nit: would it be useful to have a Patient resource whose birthdate false outside the range and validate that their birthdate was not changed?

File: src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        public async Task GivenPerturbAnonymizationConfig_WhenAnonymizeResource_ThenPerturbedNodeShouldBeReturned()
+        {
+            string configurationContent =
+            @"
+            {
+              ""fhirPathRules"": [
+                {
+                  ""path"": ""Condition.onset as Age"",
+                  ""method"": ""perturb"",
+                  ""span"": 0,
```
Comment (namadabu) [inline src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs:161]: nit: just curious whether it is normal to use `perturb` with `span: 0` since that means the value is not changed? Should we also have a test where `span` is set to a non-zero value and then we validate that the result is within the expected range?

**Review by namadabu — APPROVED** [review]
:shipit:

File: src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs (hunk 4 above)
Comment (tongwu-msft) [reply src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportAnonymizerTests.cs:161]: thanks for the comments, we want to have basic tests to cover in these tests to make sure integration with FHIR.
